hrms/hr/doctype/goal/goal_tree.js

```diff
@@ -234,32 +249,75 @@ frappe.treeview_settings["Goal"] = {
 							default: node.data.progress,
 						},
 					],
-					primary_action: function() {
+					primary_action: function () {
 						dialog.hide();
-						return frappe.call({
-							method: "hrms.hr.doctype.goal.goal.update_progress",
-							args: {
-								progress: dialog.get_values()["progress"],
-								goal: node.data.value
-							},
-						}).then(r => {
-							if (!r.exc && r.message) {
-								frappe.treeview_settings["Goal"].treeview.tree.load_children(node.parent_node, true);
+						return frappe
```

Comment:
Can be moved to a separate function to improve readability? This and marking goals and completed